odoo/addons/base/models/ir_model.py

```diff
@@ -1257,6 +1258,52 @@ def unlink(self):
 
         return result
 
+    def _process_ondelete(self):
+        """ Process the 'ondelete' of the given selection values. """
+        for selection in self.filtered('field_id.store'):
+            Model = self.env[selection.field_id.model]
+            field = Model._fields[selection.field_id.name]
+            ondelete = (field.ondelete or {}).get(selection.value)
+            if callable(ondelete):
+                ondelete(selection._get_records())
+            elif not ondelete or ondelete == 'set null':
+                selection._replace_selection(None)
+            elif ondelete == 'set default':
+                default_value = field.default(Model)
+                default_value = field.convert_to_write(default_value, Model)
+                default_value = field.convert_to_column(default_value, Model)
+                selection._replace_selection(default_value)
+            else:
+                # ondelete='cascade'
+                selection._delete_records()
+
+    def _get_records(self):
+        """ Return the records having 'self' as a value. """
+        Model = self.env[self.field_id.model]
+        query = 'SELECT id FROM "{table}" WHERE "{field}"=%s'.format(
+            table=Model._table, field=self.field_id.name,
+        )
+        self.env.cr.execute(query, [self.value])
+        return Model.browse(r[0] for r in self.env.cr.fetchall())
+
+    def _delete_records(self):
+        """ Delete the records having 'self' as a value. """
+        self.ensure_one()
+        query = 'DELETE FROM "{table}" WHERE "{field}"=%s'.format(
+            table=self.env[self.field_id.model]._table,
+            field=self.field_id.name,
+        )
+        self.env.cr.execute(query, [self.value])
+
+    def _replace_selection(self, value):
+        """ Replace the value of 'self' by 'value' on the corresponding records. """
+        self.ensure_one()
+        query = 'UPDATE "{table}" SET "{field}"=%s WHERE "{field}"=%s'.format(
+            table=self.env[self.field_id.model]._table,
+            field=self.field_id.name,
+        )
+        self.env.cr.execute(query, [value, self.value])
```

Comment:
What about triggering the recomputation of fields that depend on `field` on the updated records?
You would have that if you use an explicit call to `write()`.